### DISCUSSION TITLE
Load full content from web

### DISCUSSION THREAD

**Discussion by tonycpsu** [ViennaRSS/vienna-rss#476] [Ideas]
The Inoreader web app has a nice feature that can pull down full content from an article's URL and render it reasonably well inside the reader.  This would be great to have in Vienna for feeds that don't provide full content in the feed itself.  It'd be even better if feeds could be configured to load full content for all articles automatically, either when the feed is pulled down, or when the article is viewed.

**Comment by barijaona** [comment]
Have you tried the "Folder -> Use Web Page for Articles" menu item ?

**Comment by tonycpsu** [comment]
Yes, but I was hoping for something where it would pull the content in and render it in the style of the reader for a consistent look and feel without all the web junk (headers, ads, comment sections, etc.)  I've seen this done in some Android readers like NewsPlus, and as I said Inoreader does this when you hit the "w" key while viewing an article.

**Comment by barijaona** [comment]
I do not have time to test this, but this might be done by porting a descendant of arc90's readability (probably [Zihua Li](https://github.com/luin/readability)'s version or [Mozilla](https://github.com/mozilla/readability)'s version) as a [Vienna plugin](http://www.vienna-rss.org/?page_id=120).

**Comment by barijaona** [comment]
I played with various tools, and can assert that selecting the "full" content from a webpage is easier said than done… It will differ with type of webpage and user preference. What one user will consider as a "good" extraction will be considered "insufficient" or "bloated" by others. However, [DZReadability](https://github.com/DZamataev/DZReadability/) is my framework candidate.

**Comment by matthewruzzi** [comment]
There is an unofficial swift wrapper for @mozilla/readability. 
https://github.com/Ryu0118/swift-readability

**Comment by matthewruzzi** [comment]
We also could send Apple feedback about adding reader mode support to WKWebView.
